src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/IWPDocumentRepository.kt

```diff
@@ -1,8 +1,21 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository
 
-import org.springframework.data.repository.CrudRepository
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.IWPDocument
 
 @Repository
-interface IWPDocumentRepository : CrudRepository<IWPDocument, Long>
+interface IWPDocumentRepository : JpaRepository<IWPDocument, Long> {
+  @Query(
+    """
+      select document.id from IWPDocument document
+       left join IWPTemplate template on template.id = document.template.id
```

Comment:
does this need to be a `left` if you're restricting by `template.name` anyway?

Reply:
you're right - I will remove